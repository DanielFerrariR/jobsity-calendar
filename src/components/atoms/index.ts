export { default as Box } from './box'
export { default as Button } from './button'
export { default as Paper } from './paper'
export { default as Typography } from './typography'
export { default as TextField } from './text_field'
export { default as Container } from './container'
export { default as AppBar } from './appbar'
export { default as IconButton } from './icon_button'
export { default as Toolbar } from './toolbar'
export { default as Image } from './image'
export { default as Grid } from './grid'
export { default as Link } from './link'
export { default as Checkbox } from './checkbox'
export { default as Accordion } from './accordion'
export { default as AccordionSummary } from './accordion_summary'
export { default as AccordionDetails } from './accordion_details'
export { default as Tooltip } from './tooltip'
export { default as Modal } from './modal'
export { default as MenuItem } from './menu_item'
export { default as ListItemIcon } from './list_item_icon'
export { default as ListItemText } from './list_item_text'
export { default as ListItemSecondaryAction } from './list_item_secondary_action'
export { default as MenuList } from './menu_list'
export { default as Popover } from './popover'
export { default as Fade } from './fade'
export { default as Backdrop } from './backdrop'
export { default as CircularProgress } from './circular_progress'
export { default as LinearProgress } from './linear_progress'
export { default as Skeleton } from './skeleton'
export { default as List } from './list'
export { default as ListItem } from './list_item'
export { default as ClickAwayListener } from './click_away_listener'
export { default as Badge } from './badge'
export { default as Portal } from './portal'
export { default as Snackbar } from './snackbar'
export { default as Step } from './step'
export { default as StepContent } from './stepContent'
export { default as StepLabel } from './stepLabel'
export { default as Stepper } from './stepper'
export { default as TableBody } from './table_body'
export { default as TableContainer } from './table_container'
export { default as TableHead } from './table_head'
export { default as TableRow } from './table_row'
export { default as Table } from './table'
export { default as TableCell } from './table_cell'
export { default as DateTimePicker } from './date_time_picker'
