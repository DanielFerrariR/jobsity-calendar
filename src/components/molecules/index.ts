export { CustomTableHead, CustomTableBody } from './custom_table'
export { default as Modal } from './modal'
export { default as ReminderCard } from './reminder_card'
