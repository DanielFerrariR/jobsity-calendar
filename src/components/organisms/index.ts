export { default as Calendar } from './calendar'
export { default as ListRemindersModal } from './list_reminders_modal'
export { default as CreateReminderModal } from './create_reminder_modal'
export { default as EditReminderModal } from './edit_reminder_modal'
