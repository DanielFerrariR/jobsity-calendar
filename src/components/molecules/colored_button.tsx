import { Box } from 'src/components/atoms'
import { styled } from '@material-ui/core'

const ColoredButton = styled(Box)(({ theme }) => ({
  cursor: 'pointer',
  height: 24,
  width: 24,
  borderRadius: theme.shape.borderRadius,
  '&:hover': {
    opacity: 0.8
  }
}))

export default ColoredButton
