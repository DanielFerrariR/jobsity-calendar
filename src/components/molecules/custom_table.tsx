import React from 'react'
import { TableHead, TableBody } from 'src/components/atoms'
import { styled, TableHeadProps, TableBodyProps, fade } from '@material-ui/core'

const StyledTableHead = styled(TableHead)(({ theme }) => ({
  '& tr th': {
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.common.white,
    border: `1px solid ${theme.palette.primary.main}`,
    padding: 0
  }
}))

const StyledTableBody = styled(TableBody)(({ theme }) => ({
  '& tr td': {
    border: '1px solid black',
    verticalAlign: 'initial',
    padding: '8px',
    height: 120,
    width: 120,
    maxWidth: 120,
    '&:hover': {
      backgroundColor: fade(theme.palette.primary.main, 0.1),
      cursor: 'pointer'
    }
  },
  '& tr td:first-child': {
    backgroundColor: theme.palette.grey[200],
    '&:hover': {
      backgroundColor: fade(theme.palette.primary.main, 0.1),
      cursor: 'pointer'
    }
  },
  '& tr td:last-child': {
    backgroundColor: theme.palette.grey[200],
    '&:hover': {
      backgroundColor: fade(theme.palette.primary.main, 0.1),
      cursor: 'pointer'
    }
  }
}))

export const CustomTableHead: React.FC<TableHeadProps> = (props) => {
  return <StyledTableHead {...props} />
}

export const CustomTableBody: React.FC<TableBodyProps> = (props) => {
  return <StyledTableBody {...props} />
}
