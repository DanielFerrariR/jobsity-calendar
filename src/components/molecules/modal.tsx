import React, { Dispatch, SetStateAction } from 'react'
import { Fade, Modal as MuiModal, Paper, Backdrop } from 'src/components/atoms'

interface Props {
  open: boolean
  setOpen: Dispatch<SetStateAction<any>>
}

const Modal: React.FC<Props> = ({ children, open, setOpen }) => {
  const handleClose = () => {
    setOpen(false)
  }

  return (
    <MuiModal
      data-testid="modal"
      aria-labelledby="modal-title"
      aria-describedby="description"
      open={open}
      onClose={handleClose}
      closeAfterTransition
      BackdropComponent={Backdrop}
      BackdropProps={{
        timeout: 500
      }}
      display="flex"
      alignItems="center"
      justifyContent="center"
    >
      <Fade in={open}>
        <Paper elevation={5} width={1} maxWidth={400} p={2} mx={2}>
          {children}
        </Paper>
      </Fade>
    </MuiModal>
  )
}

export default Modal
