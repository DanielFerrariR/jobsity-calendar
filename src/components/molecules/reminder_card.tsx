import React from 'react'
import { Box, Typography, Image } from 'src/components/atoms'
import { useSelector } from 'src/store'
import { isSameDay } from 'src/utils'
import { getContrastRatio } from '@material-ui/core'

interface Props {
  date: Date
  testId: string
}

const ReminderCard: React.FC<Props> = ({ date, testId }) => {
  const reminders = useSelector((state) => state.reminders)
  const filteredReminders = reminders?.filter((element) =>
    isSameDay(element.date, date)
  )
  const sortedReminders = filteredReminders?.sort((first, second) => {
    if (first.date < second.date) {
      return -1
    }

    if (first.date > second.date) {
      return 1
    }

    return 0
  })

  return (
    <>
      {sortedReminders &&
        (sortedReminders.length < 3 ? (
          sortedReminders.map((each, index) => (
            <Box
              key={each.id}
              bgcolor={each.color}
              mb={1}
              borderRadius={4}
              px={1}
              display="flex"
              alignItems="center"
              height={32}
              data-testid={`${testId}-${index}`}
            >
              <Typography
                variant="caption"
                width={128}
                color={
                  getContrastRatio('#fff', each.color) > 4.5 ? '#fff' : '#000'
                }
                style={{
                  textOverflow: 'ellipsis',
                  overflow: 'hidden'
                }}
              >
                {each.text}
              </Typography>
              {each.weather && (
                <Image src={each.weather.icon} height={24} width={24} />
              )}
            </Box>
          ))
        ) : (
          <>
            <Box
              bgcolor={sortedReminders[0].color}
              mb={1}
              borderRadius={4}
              px={1}
              display="flex"
              alignItems="center"
              height={32}
              data-testid={`${testId}-0`}
            >
              <Typography
                variant="caption"
                width={128}
                color={
                  getContrastRatio('#fff', sortedReminders[0].color) > 4.5
                    ? '#fff'
                    : '#000'
                }
                style={{
                  textOverflow: 'ellipsis',
                  overflow: 'hidden'
                }}
              >
                {sortedReminders[0].text}
              </Typography>
              {sortedReminders[0].weather && (
                <Image
                  src={sortedReminders[0].weather.icon}
                  height={24}
                  width={24}
                />
              )}
            </Box>
            <Typography variant="caption" data-testid={`${testId}-extra`}>
              + {sortedReminders?.length - 1} more
            </Typography>
          </>
        ))}
    </>
  )
}

export default ReminderCard
