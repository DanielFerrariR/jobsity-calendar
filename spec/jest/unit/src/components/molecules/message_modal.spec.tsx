import React from 'react'
import { MessageModal } from 'src/components/molecules'
import { render } from 'spec/jest/config/test_utils'
import { axe } from 'jest-axe'

describe('testing MessageModal component', () => {
  test('should snapshot works with english language', () => {
    const { container } = render(
      <MessageModal open setOpen={(): void => {}} message="Message" />
    )

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with portuguese language', () => {
    const { container } = render(
      <MessageModal open setOpen={(): void => {}} message="Message" />,
      {
        language: 'pt'
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should snapshot works with spanish language', () => {
    const { container } = render(
      <MessageModal open setOpen={(): void => {}} message="Message" />,
      {
        language: 'es'
      }
    )

    expect(container).toMatchSnapshot()
  })

  test('should the page be accessible', async () => {
    const { container } = render(
      <MessageModal open setOpen={(): void => {}} message="Message" />
    )
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
