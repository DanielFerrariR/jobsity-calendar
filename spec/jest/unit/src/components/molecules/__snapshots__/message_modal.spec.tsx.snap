// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing MessageModal component should snapshot works with english language 1`] = `
<div
  aria-hidden="true"
>
  <div
    aria-describedby="description"
    aria-labelledby="modal-title"
    class="ForwardRef(Modal)-root ForwardRef(Modal)-root"
    data-testid="modal-alert"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root WithStyles(ForwardRef(Backdrop))-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root Component-root Component-root ForwardRef(Fade)-root MuiPaper-elevation5 MuiPaper-rounded"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="modal-message"
        id="modal-message"
      >
        Message
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        data-testid="button-message-modal-ok"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Ok
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`testing MessageModal component should snapshot works with portuguese language 1`] = `
<div
  aria-hidden="true"
>
  <div
    aria-describedby="description"
    aria-labelledby="modal-title"
    class="ForwardRef(Modal)-root ForwardRef(Modal)-root"
    data-testid="modal-alert"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root WithStyles(ForwardRef(Backdrop))-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root Component-root Component-root ForwardRef(Fade)-root MuiPaper-elevation5 MuiPaper-rounded"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="modal-message"
        id="modal-message"
      >
        Message
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        data-testid="button-message-modal-ok"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Ok
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`testing MessageModal component should snapshot works with spanish language 1`] = `
<div
  aria-hidden="true"
>
  <div
    aria-describedby="description"
    aria-labelledby="modal-title"
    class="ForwardRef(Modal)-root ForwardRef(Modal)-root"
    data-testid="modal-alert"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root WithStyles(ForwardRef(Backdrop))-root"
      style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root Component-root Component-root ForwardRef(Fade)-root MuiPaper-elevation5 MuiPaper-rounded"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <p
        class="MuiTypography-root Component-root Component-root MuiTypography-body1 MuiTypography-alignCenter"
        data-testid="modal-message"
        id="modal-message"
      >
        Message
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained Component-root Component-root MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        data-testid="button-message-modal-ok"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Ok
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</div>
`;
